### PR TITLE
When using xmlhttp when exiting, there is an assertion problem refere…

### PR DESCRIPTION
**Pull request by qiuguohua**

### CODE REVIEW

File: native/cocos/engine/BaseEngine.h [hunk 1]
```diff
@@ -104,11 +104,11 @@ class CC_DLL BaseEngine : public CallbacksInvoker,
      */
     virtual void removeEventCallback(OSEventType evtype) = 0;
 
-    using SchedulerPtr = std::shared_ptr<Scheduler>;
+    using SchedulerSharePtr = std::shared_ptr<Scheduler>;
```
Comment (dumganhar) [inline native/cocos/engine/BaseEngine.h]: Why rename this?

Comment (qiuguohua) [reply native/cocos/engine/BaseEngine.h]: Because ptr is too abstract. The specific points may be better understood

**Review by dumganhar — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Comment by suqionglian** [comment]
@cocos-robot run test cases

**Comment by suqionglian** [comment]
@cocos-robot run test cases

**Comment by suqionglian** [comment]
@cocos-robot run test cases

**Comment by suqionglian** [comment]
@cocos-robot run test cases

**Comment by suqionglian** [comment]
@cocos-robot run test cases

**Comment by suqionglian** [comment]
@cocos-robot  run test cases

**Comment by suqionglian** [comment]
@cocos-robot run test cases

**Pull request merged**